### PR TITLE
204 as a dev i want to enforce non null constraints on entity fields to prevent records with all null fields

### PR DESCRIPTION
**Pull request by Francois-Werbrouck**
ref #204 

This PR focuses on adding table constraint to not insert empty rows.

### CODE REVIEW

**Comment by Francois-Werbrouck** [comment]
image

File: fertiscan/db/bytebase/schema_0.0.15.sql [hunk 1]
```diff
@@ -60,7 +60,12 @@ IF (EXISTS (SELECT 1 FROM information_schema.schemata WHERE schema_name = 'ferti
         "website" text,
         "phone_number" text,
         "location_id" uuid REFERENCES "fertiscan_0.0.15".location(id),
-        "edited" boolean DEFAULT false
+        "edited" boolean DEFAULT false,
+        CONSTRAINT check_not_all_null CHECK (
+            (name IS NOT NULL)::integer +
+            (website IS NOT NULL)::integer +
+            (phone_number IS NOT NULL)::integer +
+            (location_id is not null)::integer >= 1
```
Comment (k-allagbe) [inline fertiscan/db/bytebase/schema_0.0.15.sql:68]: Should we leave `fertiscan_0.0.15` the way it was as there is a new schema?

File: fertiscan/db/bytebase/schema_0.0.15.sql [hunk 2]
```diff
@@ -161,12 +161,11 @@ IF (EXISTS (SELECT 1 FROM information_schema.schemata WHERE schema_name = 'ferti
     "unit_id" uuid REFERENCES "fertiscan_0.0.15".unit(id),
     "metric_type" "fertiscan_0.0.15".metric_type,
     "label_id" uuid REFERENCES "fertiscan_0.0.15".label_information(id) ON DELETE CASCADE
-    );
 
     CREATE TABLE "fertiscan_0.0.15"."sub_type" (
         "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
         "type_fr" text Unique NOT NULL,
-        "type_en" text unique NOT NULL
+        "type_en" text unique NOT NULL,
```
Comment (k-allagbe) [inline fertiscan/db/bytebase/schema_0.0.15.sql:168]: I'm not sure the trailing comma will pass.

Comment (Francois-Werbrouck) [reply fertiscan/db/bytebase/schema_0.0.15.sql]: done

**Review by k-allagbe — APPROVED** [review]

**Pull request merged**